addons/udes_stock/models/stock_picking.py

```diff
@@ -2783,130 +2784,137 @@ def raise_insufficiency_error(moves):
-            limit = 1
+            limit = 100 if reserve_all else picking_type.u_num_reservable_pickings
             processed = Picking.browse()
             by_type = lambda x: x.picking_type_id == picking_type
 
             while reserve_all or to_reserve > 0:
 
                 if self:
                     # Removed processed pickings from self
-                    pickings = self.filtered(by_type) - processed
+                    pickings_to_process = self.filtered(by_type) - processed
                 else:
                     domain = base_domain[:]
                     if processed:
                         domain.append(("id", "not in", processed.ids))
-                    pickings = Picking.search(domain, limit=limit)
+                    pickings_to_process = Picking.search(domain, limit=limit)
 
-                if not pickings:
+                if not pickings_to_process:
                     # No pickings left to process.
                     # If u_num_reservable_pickings is -1, or there are
                     # fewer available pickings that the limit, the loop must
                     # terminate here.
                     break
 
-                batch = pickings.mapped("batch_id")
-                if batch and batch.state == "draft":
-                    # Add to seen pickings so that we don't try to process
-                    # this batch again.
-                    processed |= batch.picking_ids
-                    continue
-
-                if batch and picking_type.u_reserve_batches:
-                    pickings = batch.picking_ids
-
-                # Check for available stock and add pickings which we cannot
-                # fulfill to the processed recordset. The logic here
-                # should mirror that within the savepoint below, expect here we
-                # do not have the move lines.
-                unreservable_moves = pickings._find_unreservable_moves(return_fast=bool(self))
-                if unreservable_moves:
-                    moves = pickings.mapped("move_lines")
-                    moves_todo = (
-                        moves.filtered(lambda m: m.state in ["confirmed", "draft"])
-                        - unreservable_moves
-                    )
-                    if not picking_type.u_handle_partials:
-                        # Add to processed to skip reservation.
-                        processed |= pickings
-
-                        if self:
-                            raise_insufficiency_error(unreservable_moves)
-                        continue
-                    elif not moves_todo:
-                        # Add to processed to skip reservation.
-                        processed |= pickings
+                # "pickings" is plural as the individual picking may be
+                # extended if it's in a batch.
+                for pickings in pickings_to_process:
```

Comment:
I think we should break in case to_reserve is less than 0 inside the for loop.
Currently it seems like if limit is 20 because of u_num_reservable_pickings and suppose all pickings are different batches we go in this for loop and reserve all batches one by one without breaking in case pickings reserved have passed the limit.
Maybe a while loop and to add in the while to_reserve to would be a suggestion

Reply:
Or maybe at line 2917 another if if to_reserve <= 0: break the for loop

Reply:
Fixed by e62d7c743.